typescript/infra/src/utils/safe.ts

```diff
@@ -41,6 +41,40 @@ export function createSafeTransactionData(call: CallData): MetaTransactionData {
   };
 }
 
+export async function executeTx(
+  chain: ChainNameOrId,
+  multiProvider: MultiProvider,
+  safeAddress: Address,
+  safeTxHash: string,
+): Promise<void> {
+  const { safeSdk, safeService } = await getSafeAndService(
+    chain,
+    multiProvider,
+    safeAddress,
+  );
+  let safeTransaction;
+  try {
+    safeTransaction = await safeService.getTransaction(safeTxHash);
+    if (!safeTransaction) {
+      throw new Error(`Failed to fetch transaction details for ${safeTxHash}`);
+    }
+  } catch (error) {
+    console.error(chalk.red(`Error fetching transaction: ${error}`));
```

Comment:
curious if you think the catching here / on line 69 should be left to the caller? feels useful for the caller to know if it was successful or not